### PR TITLE
fix(FRO-37): Ensure user avatar is displayed properly in the chat sec…

### PR DESCRIPTION
**Pull request by alvyynm**
# Linear Ticket

**My PR Fixes ID-FRO-37**
[Linear Ticket](https://linear.app/team-headlight/issue/FRO-37/ensure-user-avatar-is-displayed-properly-in-the-chat-section-and)

# Changes proposed

## What were you told to do?

The Avatars in Zuri Chat are not showing very well. The user avatar is not displaying and the avatars of people in the chat are not showing. I cannot change my avatar (upload issue?). People changing their avatar should reflect instantly. This was to be fixed to work perfectly.
[Slack task](https://hng9.slack.com/archives/C041S7VF5JR/p1668588834845889)

## What did you do?

- Ensure the avatars in Zuri Chat are showing up. 
- Add the ability to change the user avatar.  
- Changing avatars reflect instantly. 

# Check List (Check all the applicable boxes)

🚨Please review the [style guide for contributing](https://github.com/zurichat/zc_main/blob/dev/docs/STYLING.md) and [guidelines for contributing](https://github.com/zurichat/zc_main/blob/dev/docs/CONTRIBUTING.md) to this repository.

- [x] My code follows the code style of this project.
- [x] This PR does not contain plagiarized content.
- [x] The title and description of the PR is clear and explains the approach.
- [x] I am making a pull request against the **dev branch** (left side).
- [x] My commit messages styles matches our requested structure.
- [x] My code additions will fail neither code linting checks nor unit test.
- [x] I am only making changes to files I was requested to.

# Screenshots/ Videos

https://user-images.githubusercontent.com/40637276/204134187-1d848b02-b7bc-4a17-80f9-468a482b05f8.mov

### CODE REVIEW

**Comment by billmal071** [comment]
Add a default image for users without pictures or when they remove their pictures.

File: packages/main/src/components/protected/topbar/components/EditProfile.jsx [hunk 1]
```diff
@@ -120,7 +120,7 @@ const EditProfile = () => {
     setUserProfileImage(user.image_url);
   }, [user]);
   // This will handle the profile form submission
-  const handleFormSubmit = e => {
+  const handleFormSubmit = async e => {
     e.preventDefault();
     setState({ ...state, loading: true });
```
Comment (billmal071) [inline packages/main/src/components/protected/topbar/components/EditProfile.jsx:125]: setState((prev) => ({...prev, loading: true})

File: packages/main/src/components/protected/topbar/components/EditProfile.jsx [hunk 2]
```diff
@@ -141,21 +141,38 @@ const EditProfile = () => {
...
-        setState({ loading: false });
-        toast.error(err?.message, {
-          position: "top-center"
-        });
+    try {
+      const res = await authAxios.patch(
+        `/organizations/${orgId}/members/${user._id}/profile`,
+        data
+      );
+      console.log(res);
```
Comment (billmal071) [inline packages/main/src/components/protected/topbar/components/EditProfile.jsx:149]: Remove console.log.

File: packages/main/src/components/protected/topbar/components/EditProfile.jsx [hunk 3]
```diff
@@ -141,21 +141,38 @@ const EditProfile = () => {
...
+      setState({ loading: false });
+      toast.success("User Profile Updated Successfully", {
+        position: "top-center"
+      });
+    } catch (err) {
+      console.error("Error", err);
+      setState({ loading: false });
```
Comment (billmal071) [inline packages/main/src/components/protected/topbar/components/EditProfile.jsx:156]: This is wrong, look at the fix in the comment above.

File: packages/main/src/components/protected/topbar/components/EditProfile.jsx [hunk 4]
```diff
@@ -141,21 +141,38 @@ const EditProfile = () => {
...
+        `/organizations/${orgId}/members/${user._id}/profile`,
+        data
+      );
+      console.log(res);
+      setState({ loading: false });
+      toast.success("User Profile Updated Successfully", {
+        position: "top-center"
+      });
+    } catch (err) {
+      console.error("Error", err);
```
Comment (billmal071) [inline packages/main/src/components/protected/topbar/components/EditProfile.jsx:155]: add toast for errors.

File: packages/main/src/components/protected/topbar/components/EditProfile.jsx [hunk 5]
```diff
@@ -141,21 +141,38 @@ const EditProfile = () => {
...
-        toast.error(err?.message, {
-          position: "top-center"
-        });
+    try {
+      const res = await authAxios.patch(
+        `/organizations/${orgId}/members/${user._id}/profile`,
+        data
+      );
+      console.log(res);
+      setState({ loading: false });
```
Comment (billmal071) [inline packages/main/src/components/protected/topbar/components/EditProfile.jsx:150]: same here also, write this properly.

File: packages/main/src/components/protected/topbar/components/EditProfile.jsx [hunk 6]
```diff
@@ -141,21 +141,38 @@ const EditProfile = () => {
...
+    //       position: "top-center"
+    //     });
+    //   })
+    //   .catch(err => {
+    //     console.error(err);
+    //     setState({ loading: false });
+    //     toast.error(err?.message, {
+    //       position: "top-center"
+    //     });
+    //   });
```
Comment (billmal071) [inline packages/main/src/components/protected/topbar/components/EditProfile.jsx:175]: remove commented out code.

File: packages/main/src/components/protected/topbar/components/TopNavbar.jsx [hunk 7]
```diff
@@ -137,16 +144,30 @@ const TopNavbar = ({ toggleSidebar }) => {
...
+  // useEffect(() => {
+  //   UpdateInfo();
+  // }, [userProfileImage]);
 
-  const UpdateInfo = () => {
-    getUserInfo().then(res => {
-      setUserProfileImage(res?.user.image_url);
-      setUser(res.user);
-    });
+  // const UpdateInfo = () => {
```
Comment (billmal071) [inline packages/main/src/components/protected/topbar/components/TopNavbar.jsx:151]: Remove commented-out code, remove console.log and console.error, use toast to display the info to the users.

**Review by billmal071 — CHANGES_REQUESTED** [review]

**Comment by billmal071** [comment]
Also, I don't think I saw a check for what type of file it accepts.

**Comment by alvyynm** [comment]
> Add a default image for users without pictures or when they remove their pictures.

It allows the user to change their profile picture whenever they want. There's already a default image when you create an account.

**Comment by alvyynm** [comment]
> Also, I don't think I saw a check for what type of file it accepts.

There's already a check for image file types.
Screenshot 2022-11-27 at 17 00 37

**Comment by alvyynm** [comment]
@billmal071 I've made the changes

**Review by billmal071 — APPROVED** [review]

**Pull request merged**